perun-rpc/src/main/java/cz/metacentrum/perun/rpc/Api.java

```diff
@@ -834,6 +842,71 @@ private Deserializer selectDeserializer(String format, HttpServletRequest req) t
 		}
 	}
 
+	private record OidcSubIssuer(String sub, String issuer) {}
+
+	private static OidcSubIssuer initOidcAuth(HttpServletRequest req, Map<String, String> additionalInformations) {
+		var accessToken = req.getHeader(OIDC_ACCESS_TOKEN);
+		var restTemplate = new RestTemplate();
+		var issuer = req.getHeader(OIDC_CLAIM_ISS);
+		var config = restTemplate.getForObject(issuer + "/.well-known/openid-configuration", JsonNode.class);
+		var headers = new HttpHeaders();
+		headers.setBearerAuth(accessToken);
+		HttpEntity<Void> entity = new HttpEntity<>(headers);
+		var userInfoResponse = restTemplate.exchange(config.path("userinfo_endpoint").textValue(), HttpMethod.GET, entity, JsonNode.class);
+		var userInfo = userInfoResponse.getBody();
+		if(isNotEmpty(userInfo.path("error").asText())) {
+			log.error("Call to user info endpoint failed, the error is: {}", userInfo);
+			throw new InternalErrorException("Call to user info endpoint failed, the error is" + userInfo);
```

Comment:
This error handling is correct, maybe better reaction/message would be more appropriate